ipatests/test_integration/test_mkey_upgrade.py

```diff
@@ -0,0 +1,137 @@
+#
+# Copyright (C) 2015  FreeIPA Contributors see COPYING for license
+#
+
+from __future__ import absolute_import
+
+import textwrap
+import time
+
+from ipatests.pytest_ipa.integration import tasks
+from ipatests.test_integration.base import IntegrationTest
+
+
+class TestMkeyUpgrade(IntegrationTest):
+
+    num_replicas = 1
+    topology = 'line'
+
+    @classmethod
+    def install(cls, mh):
+        cls.master.put_file_contents(
+            '/etc/profile.d/ipaplatform.sh',
+            'export IPAPLATFORM_OVERRIDE=fedora_aes_sha1_mkey')
+        cls.master.run_command(['mkdir', '/etc/systemd/system/ipa.service.d'])
+        cls.master.put_file_contents(
+            '/etc/systemd/system/ipa.service.d/platform.conf',
+            '[Service]\n'
+            'Environment="IPAPLATFORM_OVERRIDE=fedora_aes_sha1_mkey"')
+        cls.master.run_command(['systemctl', 'daemon-reload'])
+        tasks.install_master(cls.master, setup_dns=False)
+        tasks.install_replica(cls.master, cls.replicas[0], setup_dns=False)
+
+    def test_old_active_mkey(self):
+        self.master.run_command(['env'])
+        self.master.run_command(
+            "kdb5_util list_mkeys | tee /dev/stderr | "
+            "grep -Eq '^KVNO: 1, Enctype: aes256-cts-hmac-sha1-96, .* \\*$'")
```

Comment:
You can also use `result = self.master.run_command(...)` and then parse result.stdout_text.

Reply:
Fixed.